core/storage/blog_post/gae_models.py

```diff
@@ -0,0 +1,457 @@
+        random hash of 12 chars.
+
+        Returns:
+            str. A blog post ID that is different from the IDs of all
+            the existing blog posts.
+
+        Raises:
+            Exception. There were too many collisions with existing blog post
+                IDs when attempting to generate a new blog post ID.
+        """
+
+        for _ in python_utils.RANGE(base_models.MAX_RETRIES):
+            blog_post_id = utils.convert_to_hash(
+                python_utils.UNICODE(
+                    utils.get_random_int(base_models.RAND_RANGE)),
+                base_models.ID_LENGTH)
+            if not cls.get_by_id(blog_post_id):
+                return blog_post_id
+        raise Exception(
+            'New blog post id generator is producing too many collisions.')
+
+    @classmethod
+    def create(cls, blog_post_id, author_id):
+        """Creates a new BlogPostModel entry.
+
+        Args:
+            blog_post_id: str. Blog Post ID of the newly-created blog post.
+            author_id: str. User ID of the author.
+
+        Returns:
+            BlogPostModel. The newly created BlogPostModel instance.
+
+        Raises:
+            Exception. A blog post with the given blog post ID exists already.
+        """
+        if cls.get_by_id(blog_post_id):
+            raise Exception(
+                'A blog post with the given blog post ID exists already.')
+
+        entity = cls(
+            id=blog_post_id,
+            author_id=author_id,
+            content='',
+            title='',
+            published_on=None,
+            url_fragment='',
+            tags=[],
+            thumbnail_filename=''
+        )
+        entity.update_timestamps()
+        entity.put()
+
+        return entity
+
+    @classmethod
+    def get_by_url_fragment(cls, url_fragment):
+        """Gets BlogPostModel by url_fragment. Returns None if the blog post
+        with the given url_fragment doesn't exist.
+
+        Args:
+            url_fragment: str. The url fragment of the blog post.
+
+        Returns:
+            BlogPostModel | None. The blog post model of the Blog or None if not
+            found.
+        """
+        return BlogPostModel.query(
+            datastore_services.all_of(
+                cls.url_fragment == url_fragment, cls.deleted == False) # pylint: disable=singleton-comparison
+        ).get()
+
+    @classmethod
+    def export_data(cls, user_id):
+        """Exports the data from BlogPostModel into dict format for Takeout.
+
+        Args:
+            user_id: str. The ID of the user whose data should be exported.
+
+        Returns:
+            dict. Dictionary of the data from BlogPostModel.
+        """
+
+        user_data = dict()
+        blog_post_models = cls.get_all().filter(
+            cls.author_id == user_id).fetch()
+
+        for blog_post_model in blog_post_models:
+            user_data[blog_post_model.id] = {
```

Comment:
@vojtechjelinek should we be exporting the internal model IDs, or would it be better to have a list of dicts here?